x/gamm/keeper/grpc_query.go

```diff
@@ -145,6 +145,22 @@ func (k Keeper) PoolParams(ctx context.Context, req *types.QueryPoolParamsReques
 	}
 }
 
+func (k Keeper) TotalPoolLiquidity(ctx context.Context, req *types.QueryTotalPoolLiquidityRequest) (*types.QueryTotalPoolLiquidityResponse, error) {
```

Comment:
We should adopt a dedicated `Querier` type methodology like some of the SDK modules do, s.t. these gRPC methods are not defined on the keeper. This provides for better abstraction.

Reply:
Oh nice! Isn't Querier a legacy because now we use grpc gateway? Is it still recommended to use for better abstraction?

Reply:
Oh sweet, the querier struct would solve the API surface leak thats been bothering me for age!

Reply:
Yup exactly